### DISCUSSION TITLE
How to create a filter with a field that does not exist in the model

### DISCUSSION THREAD

**Discussion by toshiki-kohno** [carltongibson/django-filter#1576] [Q&A]
How to create a filter with a field that does not exist in the model

I was creating a filter with a field "deletion" that does not exist in the User model.
```
class UsersFilter(filters.FilterSet):
    deletion = filters.BooleanFilter(name='deletion_time', lookup_expr='isnull')
    class Meta:
        model = Users
        fields = {
            'deletion'
            }
```
This was working in django-filter 1.1.0,
In the latest version, I get the following error
`'Meta.fields' must not contain non-model field names: deletion`

How can I create a field that does not exist in the model?

**Comment by ctilch** [comment]
I had the same problem today.

"GET /app/users/?search_for=username"

```
class UserFilter(django_filters.FilterSet):
    search_for = django_filters.CharFilter(field_name="name", lookup_expr="icontains")

    class Meta:
        model = User
        fields = ["name"]
```

works for me.

So please try:

````
class UsersFilter(filters.FilterSet):
    deletion = filters.BooleanFilter(name='deletion_time', lookup_expr='isnull')
    class Meta:
        model = Users
        fields = ['deletion_time']
````

**Comment by KianAnbarestani** [comment]
In newer versions of django-filter, the Meta.fields must only include actual model fields. For custom (non-model) fields like "deletion", you should remove them from Meta.fields and define them directly.
```
class UsersFilter(filters.FilterSet):
    deletion = filters.BooleanFilter(
        field_name='deletion_time',
        lookup_expr='isnull',
        label='Deleted'
    )

    class Meta:
        model = Users
        fields = []  # do not include 'deletion' here
```
### Key point: Just define the custom field directly in the class; no need to list it in Meta.fields.
